### PR TITLE
Fix S3247 FN: Support structs

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Part of #223

### CODE REVIEW

File: analyzers/its/expected/akka.net/S3247-Akka.Tests.Shared.Internals-netstandard2.0.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "Issues": [
+    {
+      "Id": "S3247",
+      "Message": "Replace this type-check-and-cast sequence with an \u0027as\u0027 and a null check.",
+      "Uri": "https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/its/Projects/akka.net/src/core/Akka.Tests.Shared.Internals/Helpers/XAssert.cs#L56",
```
Comment (sebastien-marichal) [inline analyzers/its/expected/akka.net/S3247-Akka.Tests.Shared.Internals-netstandard2.0.json:6]: Why was this not raised before?
`T` in this context is nullable so it cannot be a value type.

Is the semantic model not able to detect that `T` is a class?

**Review by sebastien-marichal — APPROVED** [review]
LGTM!
I left some educational questions.

File: analyzers/its/expected/akka.net/S3247-Akka-netstandard2.0.json [hunk 2]
```diff
@@ -29,6 +35,36 @@
       "Message": "Remove this cast and use the appropriate variable.",
       "Uri": "https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/its/Projects/akka.net/src/core/Akka/IO/UdpListener.cs#L87",
       "Location": "Line 87 Position 21-28"
+    },
+    {
+      "Id": "S3247",
+      "Message": "Replace this type-check-and-cast sequence with an \u0027as\u0027 and a null check.",
+      "Uri": "https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/its/Projects/akka.net/src/core/Akka/Serialization/ByteArraySerializer.cs#L45",
```
[the remark this answers is not in the document]
Comment (pavel-mikula-sonarsource) [reply analyzers/its/expected/akka.net/S3247-Akka-netstandard2.0.json:42]: It's not AFAICT. There's probably some gap in the symbol flags.

File: analyzers/its/expected/akka.net/S3247-Akka.Tests.Shared.Internals-netstandard2.0.json (hunk 1 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/its/expected/akka.net/S3247-Akka.Tests.Shared.Internals-netstandard2.0.json:6]: Same as above - NamedTypeSymbol check was removed on the way

**Comment by pavel-mikula-sonarsource** [comment]
@sebastien-marichal Please take one more look. I've added one more commit, because I've noticed one more place with struct filter

**Review by sebastien-marichal — APPROVED** [review]
LGTM!

**Pull request merged**